include/nbl/builtin/hlsl/bit.hlsl

```diff
@@ -0,0 +1,55 @@
+#ifndef _NBL_BUILTIN_HLSL_BIT_INCLUDED_
+#define _NBL_BUILTIN_HLSL_BIT_INCLUDED_
+
+#include <nbl/builtin/hlsl/cpp_compat.hlsl>
+
+
+#ifndef __HLSL_VERSION
+#include <bit>
+#else
+namespace nbl
+{
+namespace hlsl
+{
+
+uint32_t rotl(NBL_CONST_REF_ARG(uint32_t) x, NBL_CONST_REF_ARG(uint32_t) s)
```

Comment:
its templated in C++, should be templated here too

Reply:
btw `s` is supposed to be `int32_t`

btw don't use `NBL...REF_ARG(T)` on types smaller than 64bit, just use `[const] T`